### PR TITLE
Notify if an explicitly excluded folder is created

### PR DESCRIPTION
**Pull request by ckamm**
See #6222 
@SamuAlfageme

### CODE REVIEW

**Comment by ogoffart** [comment]
Don't we get also this notification when subfolders or files within the folder are removed. In that case, it would be misleading.

We could check if it is in the DB or not maybe, to avoid such false warning.

File: src/gui/folder.cpp [hunk 1]
```diff
@@ -965,6 +967,28 @@ void Folder::slotNextSyncFullLocalDiscovery()
     _timeSinceLastFullLocalDiscovery.invalidate();
 }
 
+void Folder::warnOnNewExcludedFolder(const QStringRef &path)
+{
+    // Note: This assumes we're getting file watcher notifications
+    // for folders only on creation and deletion.
+    QFileInfo fi(_canonicalLocalPath + path);
+    if (!fi.isDir())
```
Comment (ogoffart) [inline src/gui/folder.cpp]: A file would have the same problem.

**Comment by ckamm** [comment]
@ogoffart At least on linux our removal notifications point to the full path that was removed, not the parent folder. I'm up for adding the db check as a safeguard anyway - possibly the win or osx watchers behave differently.

I used the `isDir` check to limit the number of cases in which we'd construct the selective sync exclude list from the database since I was worrying that doing this for each notification might create a new bottleneck. What's your intuition regarding performance?

**Comment by ogoffart** [comment]
Looks good to me from the code point of view.
But this should be tested to find out if there are not too many unwanted warnings.

Is this warning really necessary when in any case it should show an error in the ui?

**Comment by ckamm** [comment]
@ogoffart I agree this could use testing on osx and windows. @SamuAlfageme ?

The rationale provided for the warning makes some sense (https://github.com/owncloud/client/issues/6222#issuecomment-350246873) even if the client usually doesn't warn about everything that goes on that isn't synced. Making an exception for explicitly selective-sync excluded folders is ok with me.

**Comment by ckamm** [comment]
@SamuAlfageme Rebased to allow for easy testing.

**Comment by SamuAlfageme** [comment]
@ckamm awesome - I'll get to it

**Comment by SamuAlfageme** [comment]
After some time using this patch, never managed to get unwanted notifications - just the ones @ckamm mentioned in https://github.com/owncloud/client/pull/6377#issuecomment-368885780 (i.e. https://github.com/owncloud/client/issues/6222) - I'll say it's safe to merge here.

As a rule of thumb I'd say we should notify (at least once) of everything that goes in the "Issues" tab. Maybe requires https://github.com/owncloud/client/issues/5493 tho.

**Pull request merged**